include/CXXGraph/Graph/Graph_impl.hpp

```diff
@@ -232,8 +232,11 @@ void Graph<T>::removeEdge(const CXXGraph::id_t edgeId) {
 template <typename T>
 void Graph<T>::removeNode(const std::string &nodeUserId) {
   auto nodeOpt = getNode(nodeUserId);
-  auto isolatedNodeIt = isolatedNodesSet.find(nodeOpt.value());
-
+  auto isolatedNodeIt = isolatedNodesSet.end();
+  if (nodeOpt) {
+    isolatedNodeIt  = isolatedNodesSet.find(nodeOpt.value());
+  }
+  
```

Comment:
This seems ok, but it would be more concise if the entire downstream logic were gated by `nodeOpt`, i.e.:
```
auto nodeOpt = getNode(nodeUserId);

if (nodeOpt)
{
  auto isolatedNodeIt = isolatedNodesSet.find(nodeOpt.value());

  if (isolatedNodeIt != isolatedNodesSet.end()) {
    // Remove the isolated node
     ...
  } else {
    // Remove the node and connected edges
     ...
  }
}
```

Reply:
Thank you for your feedback, I made a commit with the suggested edits - let me know if it looks okay.

Reply:
Looks good. Approved